### PR TITLE
Hook form integration

### PR DESCRIPTION
**Pull request by thebesson**
# Purpose of PR

Implement the new format on the "React Form Hook Integration" documentation page.

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/website/content/integrations/react-hook-form.mdx [hunk 1]
```diff
@@ -1,189 +1,44 @@
 ---
 title: 'React Hook Form'
 slug: '/integrations/react-hook-form/'
-section: 'integrations'
+v4Doc: true
```
Comment (suevalov) [inline packages/website/content/integrations/react-hook-form.mdx:4]: We need to have `section: 'integrations'` here, and `v4Doc` is not needed

File: packages/website/content/integrations/react-hook-form.mdx [hunk 2]
```diff
@@ -4,186 +4,41 @@ slug: '/integrations/react-hook-form/'
...
-// ControlledTextInput.jsx
-import { useController } from 'react-hook-form';
-import { TextInput } from '@contentful/f36-forms';
+### Example of integration with `register()`
 
-export function ControlledTextInput(props) {
-  const {
-    field: { ref, ...inputProps },
-  } = useController(props);
+```jsx file=examples/RegisterIntegration.tsx
```
Comment (thebesson) [inline packages/website/content/integrations/react-hook-form.mdx:34]: @suevalov, how should I import 'react-hook-form'? It works in the playground, but the live preview on the documentation page doesn't see this package, there is the error `ReferenceError: useForm is not defined`
https://forma-36-git-hook-form-integration-ctfl.vercel.app/integrations/react-hook-form

File: packages/forma-36-tokens/package.json [hunk 3]
```diff
@@ -12,7 +12,8 @@
   "devDependencies": {
     "fs-extra": "^10.0.0",
     "globby": "^10.0.1",
-    "lodash": "4.17.21"
+    "lodash": "4.17.21",
+    "react-hook-form": "7.22.5"
```
Comment (suevalov) [inline packages/forma-36-tokens/package.json:16]: Why do we need this library in `forma-36-tokens/package.json`? Did you intend to add it somewhere else?

File: packages/website/components/SandpackRenderer.tsx [hunk 4]
```diff
@@ -50,11 +50,13 @@ export function SandpackRenderer(props: {
           react: '^17.0.0',
           'react-dom': '^17.0.0',
           'react-scripts': '^4.0.0',
+          '@contentful/f36-forms': 'latest',
```
Comment (suevalov) [inline packages/website/components/SandpackRenderer.tsx:53]: All components you need should be available in `@contentful/f36-components` so I don't see a reason to add `forms` package here

File: packages/website/content/integrations/examples/RegisterIntegration.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  FormControl,
+  Radio,
+  Select,
+  Textarea,
+  TextInput,
+} from '@contentful/f36-forms';
+import { Button } from '@contentful/f36-components';
+
+export default function ReactHookForm() {
+  // Call the useForm hook from react-hook-form
```
Comment (suevalov) [inline packages/website/content/integrations/examples/RegisterIntegration.tsx:16]: not sure if we need this comment here

**Review by suevalov — CHANGES_REQUESTED** [review]
I left few comment that are needed to be addressed

File: packages/forma-36-tokens/package.json (hunk 3 above)
Comment (thebesson) [reply packages/forma-36-tokens/package.json:16]: thanks for catching this

**Review by suevalov — APPROVED** [review]

File: packages/website/components/LiveEditor/ComponentSource.tsx [hunk 6]
```diff
@@ -34,6 +35,8 @@ const liveProviderScope = {
   useRef,
   useMemo,
   useContext,
+  useForm,
+  useController,
```
Comment (bgutsol) [inline packages/website/components/LiveEditor/ComponentSource.tsx:39]:
```suggestion
  // other
  useForm,
  useController,
```

File: packages/website/content/integrations/examples/RegisterIntegration.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  Form,
+  FormControl,
+  Radio,
+  Select,
+  Textarea,
+  TextInput,
+} from '@contentful/f36-forms';
+import { Button } from '@contentful/f36-components';
+
+export default function ReactHookForm() {
```
Comment (bgutsol) [inline packages/website/content/integrations/examples/RegisterIntegration.tsx:15]: 💅 Could you add please `Example` to the file name as well, just for the sake of consistency.
```suggestion
export default function RegisterIntegrationExample() {
```

File: packages/website/content/integrations/examples/UseControllerIntegration.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { useForm, useController } from 'react-hook-form';
+import { TextInput, Form, FormControl } from '@contentful/f36-forms';
+import { Button } from '@contentful/f36-components';
+
+export default function ReactHookForm() {
```
Comment (bgutsol) [inline packages/website/content/integrations/examples/UseControllerIntegration.tsx:6]: 💅 Could you add please `Example` to the file name as well, just for the sake of consistency.
```suggestion
export default function UseControllerIntegrationExample() {
```

File: packages/website/content/integrations/react-hook-form.mdx [hunk 9]
```diff
@@ -4,186 +4,41 @@ slug: '/integrations/react-hook-form/'
 section: 'integrations'
 ---
 
+import { Heading, Stack, TextLink } from '@contentful/f36-components';
+import { ExternalLinkIcon } from '@contentful/f36-icons';
+import { Props } from '@contentful/f36-docs-utils';
```
Comment (bgutsol) [inline packages/website/content/integrations/react-hook-form.mdx:9]: MDX imports don't work in the new nextJS website, it won't cause the build failure, but they are redundant.

Comment (thebesson) [reply packages/website/content/integrations/react-hook-form.mdx:9]: got it, thanks

File: packages/website/content/integrations/react-hook-form.mdx [hunk 10]
```diff
@@ -7,183 +7,34 @@ section: 'integrations'
...
-```jsx static=true
-// ControlledTextInput.jsx
-import { useController } from 'react-hook-form';
-import { TextInput } from '@contentful/f36-forms';
-
-export function ControlledTextInput(props) {
-  const {
-    field: { ref, ...inputProps },
-  } = useController(props);
+```jsx file=examples/RegisterIntegration.tsx
```
Comment (bgutsol) [inline packages/website/content/integrations/react-hook-form.mdx:30]:
```suggestion
```jsx file=examples/RegisterIntegrationExample.tsx
```

File: packages/website/content/integrations/react-hook-form.mdx [hunk 11]
```diff
@@ -7,183 +7,34 @@ section: 'integrations'
...
-        />
+First, it's necessary to create a controlled component which will use the `useController` hook and pass its props to the Forma 36 component. Then you use this controlled component in your form:
 
-        {errors.textInput && (
-          <FormControl.ValidationMessage>
-            This field is required
-          </FormControl.ValidationMessage>
-        )}
-      </FormControl>
+```jsx file=examples/UseControllerIntegration.tsx
```
Comment (bgutsol) [inline packages/website/content/integrations/react-hook-form.mdx:38]:
```suggestion
```jsx file=examples/UseControllerIntegrationExample.tsx
```

**Review by bgutsol — APPROVED** [review]

**Pull request merged**